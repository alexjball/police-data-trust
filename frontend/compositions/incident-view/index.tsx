import IncidentViewHeader from "./incident-view-header"
import IncidentBody from "./incident-view-body"
import type { Incident } from "../../helpers/api"

export default function IncidentView(incident: Incident) {
  return (
    <>
      <IncidentViewHeader {...incident} />
      <IncidentBody {...incident} />
    </>
  )
}
