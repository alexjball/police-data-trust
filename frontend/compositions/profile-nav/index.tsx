import * as React from "react"
import { ProfileMenu } from "../../models/profile"
import styles from "./profile-nav.module.css"

interface ProfileNavProps {
  currentItem: ProfileMenu
  selectNav: Function
}

const menuItems = [
  { item: ProfileMenu.USER_INFO, text: "User Information" },
  { item: ProfileMenu.PROFILE_TYPE, text: "Profile Type" },
  { item: ProfileMenu.SAVED_RESULTS, text: "Saved Results" },
  { item: ProfileMenu.SAVED_SEARCHES, text: "Saved Searches" }
]

export default function ProfileNav({ currentItem, selectNav }: ProfileNavProps) {
  const clsName = (itm: string) => (itm === currentItem ? styles.selectedItem : styles.menuItem)

  return (
    <nav className={styles.leftMenu}>
      <ul>
        {menuItems.map((mnu) => (
          <li key={mnu.item} onClick={() => selectNav(mnu.item)} className={clsName(mnu.item)}>
            {mnu.text}
          </li>
        ))}
      </ul>
    </nav>
  )
}
