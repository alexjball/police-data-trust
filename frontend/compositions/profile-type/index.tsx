import * as React from "react"
import Link from "next/link"
import { profileTypeContent, publicUser, UserDataType } from "../../models/profile"
import { useAuth } from "../../helpers"
import styles from "./profile-type.module.css"

export default function ProfileType() {
  const { user } = useAuth()
  const userData = publicUser(user)
  const { title, content, linkText, linkPath } = profileTypeContent[userData.role]
  const { profileData, sectionTitle, profileTypeText, profileTypeLink, description } = styles
  return (
    <div className={profileData}>
      <header className={sectionTitle}>Your Profile Type</header>
      <div className={profileTypeText}>{title}</div>
      <p className={description}>{content}</p>
      <div className={profileTypeLink}>
        <Link href={linkPath}>
          <a>{linkText}</a>
        </Link>
      </div>
    </div>
  )
}

// can be whoever
export function MockProfileType({ role }: UserDataType) {
  const { title, content, linkText, linkPath } = profileTypeContent[role]
  const { profileData, sectionTitle, profileTypeText, profileTypeLink, description } = styles
  return (
    <div className={profileData}>
      <header className={sectionTitle}>Your Profile Type</header>
      <div className={profileTypeText}>{title}</div>
      <p className={description}>{content}</p>
      <div className={profileTypeLink}>
        <Link href={linkPath}>
          <a>{linkText}</a>
        </Link>
      </div>
    </div>
  )
}
