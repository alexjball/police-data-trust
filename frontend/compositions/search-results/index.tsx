import SearchResultsTable from "./search-results"
export default SearchResultsTable
