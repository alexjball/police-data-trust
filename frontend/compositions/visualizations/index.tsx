import Map from "./map"
import useSearchData from "./map/useSearchData"

export { Map, useSearchData }
