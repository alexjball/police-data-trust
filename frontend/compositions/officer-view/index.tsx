import OfficerHeader from "./officer-view-header"
import OptionalOfficerInfo from "./optional-officer-info"
import OfficerWorkHistory from "./officer-work-history"
import OfficerAffiliations from "./officer-affiliations"
import { OfficerRecordType } from "../../models/officer"
import { DataTable } from "../../shared-components/data-table/data-table"
import { resultsColumns } from "../search-results/search-results"
import { EXISTING_TEST_INCIDENTS } from "../../helpers/api/mocks/data"

export default function OfficerView(officer: OfficerRecordType) {
  return (
    <>
      <OfficerHeader {...officer} />
      <hr />
      <OptionalOfficerInfo {...officer} />
      <hr />
      <OfficerWorkHistory {...officer} />
      <OfficerAffiliations {...officer} />
      {/* TODO: <DataTable tableName='Mock Table' columns={resultsColumns} data={EXISTING_TEST_INCIDENTS} /> */}
    </>
  )
}
