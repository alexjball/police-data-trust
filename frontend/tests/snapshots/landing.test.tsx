import * as React from "react"
import { LandingPage } from "../../compositions/"
import renderer from "react-test-renderer"

it("renders Saved Searches page correctly", () => {
  const tree = renderer.create(<LandingPage />).toJSON()
  expect(tree).toMatchSnapshot()
})
