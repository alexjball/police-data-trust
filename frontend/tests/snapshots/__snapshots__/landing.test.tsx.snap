// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Saved Searches page correctly 1`] = `
<div
  className="wrapper"
>
  <div
    className="header"
  >
    <img
      alt="A blue circular logo design reads 'National Police Data Coalition'"
      className="logo medium"
      src="/NPDCLogo.svg"
    />
    <button
      className="primaryButton"
      style={
        Object {
          "backgroundColor": "white",
          "border": "#303463 thin solid",
          "color": "#303463",
          "fontWeight": "bold",
          "margin": "2rem 2rem 0 0",
        }
      }
    >
      Donate
    </button>
  </div>
  <div
    className="content"
  >
    <h1
      className="title"
    >
      Data to keep our communities safer
    </h1>
    <p
      className="text"
    >
      Police brutality thrives in anonymity.
    </p>
    <p
      className="text"
    >
      For too long, records of police violence have been obscure, inacessible, and incomplete.
    </p>
    <p
      className="emphasis"
    >
      We're Changing That.
    </p>
  </div>
  <button
    className="primaryButton"
    onClick={[Function]}
    type="submit"
  >
    Join Us
  </button>
  <a
    href="#"
  >
    Learn More
  </a>
</div>
`;
